### PR TITLE
test(remote_wal): add sqlness with kafka wal 

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

1. Add kafka wal config to sqlness 
2. Add sqlness with kafka wal ci config

#### Setup kafka standalone
1. Go to workdir
```bash
cd tests-integration/fixtures/kafka/
```
2. Run
```
docker compose -f docker-compose-standalone.yml up kafka -d
```
#### Run the Sqlness with kafka wal 
```bash
cargo sqlness -w kafka -k localhost:9092
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [ ]  This PR does not require documentation updates.

### CODE REVIEW

File: tests-integration/fixtures/kafka/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+```bash
+docker compose -f docker-compose-standalone.yml ps
+```
+
+## Stops the standalone kafka 
+```bash
+docker compose -f docker-compose-standalone.yml stop kafka 
+```
+
+## Stops and removes a standalone kafka
```
Comment (WenyXu) [inline tests-integration/fixtures/kafka/README.md:16]:
```suggestion
## Stops and removes the standalone kafka 
```

File: tests/runner/src/env.rs [hunk 2]
```diff
@@ -447,13 +460,28 @@ struct GreptimeDBContext {
...
+        }
+    }
+
+    fn is_raft_engine(&self) -> bool {
+        matches!(self.wal, WalConfig::RaftEngine)
+    }
+
+    fn kafka_wal_broker_endpoints(&self) -> String {
+        match &self.wal {
+            WalConfig::RaftEngine => String::new(),
```
Comment (MichaelScofield) [inline tests/runner/src/env.rs:481]: better to fail explicility:
```suggestion
            WalConfig::RaftEngine => unreachable!(),
```

File: tests/runner/src/main.rs [hunk 3]
```diff
@@ -63,6 +78,18 @@ async fn main() {
...
         .unwrap();
-    let runner = Runner::new(config, Env::new(data_home, args.server_addr));
+
+    let wal = match args.wal {
+        Wal::RaftEngine => WalConfig::RaftEngine,
+        Wal::Kafka => WalConfig::Kafka {
+            broker_endpoints: args
+                .kafka_wal_broker_endpoints
+                .split(',')
+                .map(|s| s.to_string())
```
Comment (MichaelScofield) [inline tests/runner/src/main.rs:88]: "s" is better to be "`trim`med"

**Review by MichaelScofield — APPROVED** [review]

File: tests/runner/src/env.rs (hunk 2 above)
Comment (WenyXu) [reply tests/runner/src/env.rs:481]: 🥲 It always calls `kafka_wal_broker_endpoints. ` no matter the WAL type.

**Review by fengjiachun — APPROVED** [review]
LGTM

**Review by killme2008 — APPROVED** [review]
LGTM 👍

**Comment by killme2008** [comment]
Looks like the ci `Sqlness Test with Kafka Wal` fails sometimes. @WenyXu

**Comment by WenyXu** [comment]
> Looks like the ci `Sqlness Test with Kafka Wal` fails sometimes. @WenyXu

Yes, I'm trying to fix it.

**Review by niebayes — APPROVED** [review]

**Pull request merged**